public/app/features/dashboard-scene/serialization/buildNewDashboardSaveModel.ts

```diff
@@ -1,7 +1,37 @@
-import { defaultDashboard } from '@grafana/schema';
+import { config } from '@grafana/runtime';
+import { VariableModel, defaultDashboard } from '@grafana/schema';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
 import { DashboardDTO } from 'app/types';
 
-export function buildNewDashboardSaveModel(urlFolderUid?: string): DashboardDTO {
+export async function buildNewDashboardSaveModel(urlFolderUid?: string): Promise<DashboardDTO> {
+  let variablesList = defaultDashboard.templating?.list;
+
+  if (config.featureToggles.filtersOnByDefault) {
+    // Add filter and group by variables if the datasource supports it
+    const defaultDs = await getDatasourceSrv().get();
+
+    if (defaultDs.getTagKeys) {
+      const datasourceRef = {
+        type: defaultDs.meta.id,
+        uid: defaultDs.uid,
+      };
+
+      const fitlerVariable: VariableModel = {
+        datasource: datasourceRef,
+        name: 'Filter',
+        type: 'adhoc',
+      };
+
+      const groupByVariable: VariableModel = {
+        datasource: datasourceRef,
+        name: 'Group by',
+        type: 'groupby',
+      };
```

Comment:
wonder if a separate feature toggle should control this one, as it's so different from the filters one (and not supported by any core data source).

Ideally it would be nice users / admins could save a dashboard and have it be used as the default "new" dashboard.